db/mongo_test.go

```diff
@@ -0,0 +1,302 @@
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron3 = CronJob{
+	ID:       "",
+	IsActive: false,
+	Function: "test2",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron4 = CronJob{
+	ID:       "",
+	IsActive: true,
+	Function: "test3",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron5 = CronJob{
+	ID:       bson.NewObjectId().Hex(),
+	IsActive: true,
+	Function: "test4",
+	Workload: "",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+var cron6 = CronJob{
+	ID:       bson.NewObjectId().Hex(),
+	IsActive: true,
+	Function: "test4",
+	Workload: "{\"foo\":\"bar\"}",
+	CronTime: "6 5 4 3 2 1",
+	TimeZone: trialTimeZone,
+	Created:  trialCreateTime,
+}
+
+func TestParseWorkload(t *testing.T) {
+	w1 := "--task"
+	assert.Equal(t, w1, parseWorkload(w1))
+
+	w2 := "{\"foo\": \"bar\"}"
+	expected := bson.M(map[string]interface{}{"foo": "bar"})
+	assert.Equal(t, expected, parseWorkload(w2))
+}
+
+func TestToCronJob(t *testing.T) {
+	mc1 := mongoCronJob{
+		ID:       bson.ObjectIdHex(cron5.ID),
+		IsActive: cron5.IsActive,
+		Function: cron5.Function,
+		Workload: "",
+		CronTime: cron5.CronTime,
+		TimeZone: cron5.TimeZone,
+		Created:  cron5.Created,
+	}
+
+	assert.Equal(t, mc1.toCronJob(), cron5)
+
+	mc2 := mongoCronJob{
+		ID:       bson.ObjectIdHex(cron6.ID),
+		IsActive: cron6.IsActive,
+		Function: cron6.Function,
+		Workload: bson.M(map[string]interface{}{"foo": "bar"}),
+		CronTime: cron6.CronTime,
+		TimeZone: cron6.TimeZone,
+		Created:  cron6.Created,
+	}
+	assert.Equal(t, mc2.toCronJob(), cron6)
+
+}
+
+func TestToMongoCron(t *testing.T) {
+	assert.Equal(t, cron1.ID, "")
+	mc1 := cron1.toMongoCronJob()
+	var id bson.ObjectId
+	expected := mongoCronJob{
+		ID:       id,
+		IsActive: true,
+		Function: "test1",
+		Workload: "",
+		CronTime: "* * * * * *",
+		TimeZone: trialTimeZone,
+		Created:  trialCreateTime,
+	}
+	assert.Equal(t, mc1, expected)
+
+	assert.NotEqual(t, cron5.ID, "")
+	mc2 := cron5.toMongoCronJob()
+	expected = mongoCronJob{
+		ID:       bson.ObjectId(mc2.ID),
+		IsActive: true,
+		Function: "test4",
+		Workload: "",
+		CronTime: "6 5 4 3 2 1",
+		TimeZone: trialTimeZone,
+		Created:  trialCreateTime,
+	}
+	assert.Equal(t, mc2, expected)
+}
+
+func TestAddJob(t *testing.T) {
+	dbName := "testAddJob"
+	testURL := mongoTestURL(dbName)
+	database, dialErr := NewMongoDB(testURL, dbName)
+	assert.NoError(t, dialErr)
+	defer database.session.Close()
+	defer dropDatabase(testURL)
+
+	addErr := database.AddJob(cron1)
+	assert.NoError(t, addErr)
+
+	var mc mongoCronJob
+	cronCollection := database.getCronCollection(nil)
+	queryErr := cronCollection.Find(bson.M{"function": cron1.Function}).One(&mc)
+	assert.NoError(t, queryErr)
+
+	cron1.ID = mc.ID.Hex()
+	cron1.Created = mc.Created
+	assert.Equal(t, cron1, mc.toCronJob())
+	cron1.ID = ""
```

Comment:
this test seems redundant to `TestGetJobs`--it does an Add followed by basically a Get. I would remove it